### ISSUE TITLE
Keras fails silently if the name of the output is not correct when using metrics.

### ISSUE THREAD

**Issue by gabrieldemarmiesse** [keras-team/keras#12068]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

```python
from keras import Model
from keras.layers import Input, Dense

input_tensor = Input((8,))
x = Dense(4)(input_tensor)
model = Model(input_tensor, x)

model.compile('sgd', 'mse', metrics={'random_inexistant_output_name': 'mae'})
# the previous line should throw an error listing the output names available and 
# printing the output name that  the user provided. It currently fails silently.
```

We should have an error message. Currently, the training works and the metric is just ignored.
